data/src/test/java/org/apache/iceberg/data/TestPartitionStatsHandler.java

```diff
@@ -0,0 +1,582 @@
+import org.apache.iceberg.Parameter;
+import org.apache.iceberg.ParameterizedTestExtension;
+import org.apache.iceberg.Parameters;
+import org.apache.iceberg.PartitionData;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.PartitionStatisticsFile;
+import org.apache.iceberg.PartitionStats;
+import org.apache.iceberg.PartitionStatsUtil;
+import org.apache.iceberg.Partitioning;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.SortOrder;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableProperties;
+import org.apache.iceberg.TestHelpers;
+import org.apache.iceberg.TestTables;
+import org.apache.iceberg.deletes.PositionDelete;
+import org.apache.iceberg.expressions.Literal;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.io.OutputFile;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.types.Comparators;
+import org.apache.iceberg.types.Types;
+import org.assertj.core.api.Assumptions;
+import org.assertj.core.groups.Tuple;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+
+@ExtendWith(ParameterizedTestExtension.class)
+public class TestPartitionStatsHandler {
+  private static final Schema SCHEMA =
+      new Schema(
+          optional(1, "c1", Types.IntegerType.get()),
+          optional(2, "c2", Types.StringType.get()),
+          optional(3, "c3", Types.StringType.get()));
+
+  protected static final PartitionSpec SPEC =
+      PartitionSpec.builderFor(SCHEMA).identity("c2").identity("c3").build();
+
+  @TempDir public File temp;
+
+  private static final Random RANDOM = ThreadLocalRandom.current();
+
+  @Parameters(name = "fileFormat = {0}")
+  public static List<Object> parameters() {
+    return Arrays.asList(FileFormat.PARQUET, FileFormat.ORC, FileFormat.AVRO);
+  }
+
+  @Parameter private FileFormat format;
+
+  @Test
+  public void testPartitionStatsOnEmptyTable() throws Exception {
+    Table testTable = TestTables.create(tempDir("empty_table"), "empty_table", SCHEMA, SPEC, 2);
+    assertThat(PartitionStatsHandler.computeAndWriteStatsFile(testTable)).isNull();
+  }
+
+  @Test
+  public void testPartitionStatsOnEmptyBranch() throws Exception {
+    Table testTable = TestTables.create(tempDir("empty_branch"), "empty_branch", SCHEMA, SPEC, 2);
+    testTable.manageSnapshots().createBranch("b1").commit();
+    PartitionStatisticsFile partitionStatisticsFile =
+        PartitionStatsHandler.computeAndWriteStatsFile(testTable, "b1");
+    // creates an empty stats file since the dummy snapshot exist
+    assertThat(partitionStatisticsFile.fileSizeInBytes()).isEqualTo(0L);
```

Comment:
test would be broken if default format changes, for example with avro format non-zero file would be created

Reply:
But it won't be flaky. So, we can update the test if the behavior changes. This is as per current behavior.

Reply:
true, I've already added +1

Reply:
Thanks for your reviews. I hope we ship this feature soon and glad to know Hive, Trino are waiting for this feature.

Reply:
I think it is bad practice to check in tests for things which are not a requirement, just "coincidentally" happens.
Could we check that the file could be read and actually emtpy?

Reply:
Thinking more on it, behavior should be same as empty table testcase above. So, will update to not throw exceptions in this case.